contextproject-ms3/src/main/java/contextproject/audio/ComplexOnsetDetector.java

```diff
@@ -0,0 +1,217 @@
+package contextproject.audio;
+
+/*
+*      _______                       _____   _____ _____  
+*     |__   __|                     |  __ \ / ____|  __ \ 
+*        | | __ _ _ __ ___  ___  ___| |  | | (___ | |__) |
+*        | |/ _` | '__/ __|/ _ \/ __| |  | |\___ \|  ___/ 
+*        | | (_| | |  \__ \ (_) \__ \ |__| |____) | |     
+*        |_|\__,_|_|  |___/\___/|___/_____/|_____/|_|     
+*                                                         
+* -------------------------------------------------------------
+*
+* TarsosDSP is developed by Joren Six at IPEM, University Ghent
+*  
+* -------------------------------------------------------------
+*
+*  Info: http://0110.be/tag/TarsosDSP
+*  Github: https://github.com/JorenSix/TarsosDSP
+*  Releases: http://0110.be/releases/TarsosDSP/
+*  
+*  TarsosDSP includes modified source code by various authors,
+*  for credits and info, see README.
+* 
+*/
+
+import be.tarsos.dsp.onsets.*;
+import be.tarsos.dsp.AudioEvent;
+import be.tarsos.dsp.AudioProcessor;
+import be.tarsos.dsp.util.PeakPicker;
+import be.tarsos.dsp.util.fft.FFT;
+import be.tarsos.dsp.util.fft.HannWindow;
+
+/**
+ * A complex Domain Method onset detection function
+ * 
+ * Christopher Duxbury, Mike E. Davies, and Mark B. Sandler. Complex domain
+ * onset detection for musical signals. In Proceedings of the Digital Audio
+ * Effects Conference, DAFx-03, pages 90-93, London, UK, 2003
+ * 
+ * The implementation is a translation of onset.c from Aubio, Copyright (C)
+ * 2003-2009 Paul Brossier <piem@aubio.org>
+ * 
+ * @author Joren Six
+ * @author Paul Brossiers
+ */
+public class ComplexOnsetDetector implements AudioProcessor, OnsetDetector{
```

Comment:
Could you add to the comment which parts are ours and which parts are from TarsosDSP.